src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useContext, useMemo } from 'react';
+import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import { TodoItem } from '../Todo';
+import { Todo } from '../../types/Todo';
+import { UserIdContext } from '../../utils/context';
+import { ErrorMessage } from '../../types/ErrorMessage';
+import { ChangeFunction } from '../../types/ChangeFunction';
+
+type Props = {
+  todos: Todo[];
+  counterActiveTodos: number;
+  creatingTodoTitle: string;
+  isClearCompleted: boolean;
+  isAllToggled: boolean;
+  showError: (errorType: ErrorMessage) => void;
+  hideError: () => void;
+  DeleteTodo: (todoId: number) => void;
+  ChangeTodo: ChangeFunction;
+};
+
+export const TodoList: React.FC<Props> = React.memo(({
+  todos,
+  counterActiveTodos,
+  creatingTodoTitle,
+  isClearCompleted,
+  DeleteTodo,
+  ChangeTodo,
+  showError,
+  hideError,
+  isAllToggled,
+}) => {
+  const userId = useContext(UserIdContext);
+
+  const creatingTodo = useMemo(
+    () => ({
+      id: 0,
+      userId,
+      title: creatingTodoTitle,
+      completed: false,
+    }),
+    [userId, creatingTodoTitle],
+  );
+
+  const isLoading = (isTodoCompleted: boolean): boolean => {
```

Comment:
`isLoading` - not the best name for function
function name should contain some action, `getIsLoading`, for example